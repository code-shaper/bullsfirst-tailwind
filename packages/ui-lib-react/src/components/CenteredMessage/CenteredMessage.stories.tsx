import { Story, Meta } from '@storybook/react';
import { ViewVerticalContainer } from '../Containers';
import { SimpleHeader } from '../Header';
import { CenteredMessage } from './CenteredMessage';

export default {
  title: 'Components/CenteredMessage',
  component: CenteredMessage,
  parameters: {
    layout: 'fullscreen',
  },
} as Meta;

const Template: Story = (args) => {
  return (
    <ViewVerticalContainer>
      <SimpleHeader>Centered Message</SimpleHeader>
      <CenteredMessage>{args.children}</CenteredMessage>
    </ViewVerticalContainer>
  );
};

export const CenteredMessageStory = Template.bind({});
CenteredMessageStory.storyName = 'CenteredMessage';
CenteredMessageStory.args = {
  children: 'Hello World!',
};
