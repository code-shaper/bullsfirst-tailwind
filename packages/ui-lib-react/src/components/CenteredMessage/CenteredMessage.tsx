import { CenteredContainer } from '../Containers';
import { DivProps } from '../ReactProps';

export function CenteredMessage({ children, ...props }: DivProps) {
  return (
    <CenteredContainer className="p-6" {...props}>
      <h1 className="text-2xl text-neutral-default">{children}</h1>
    </CenteredContainer>
  );
}
