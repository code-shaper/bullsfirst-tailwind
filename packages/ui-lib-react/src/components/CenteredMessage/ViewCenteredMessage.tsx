import { ViewCenteredContainer } from '../Containers';
import { DivProps } from '../ReactProps';

export function ViewCenteredMessage({ children, ...props }: DivProps) {
  return (
    <ViewCenteredContainer className="p-6" {...props}>
      <h1 className="text-2xl text-neutral-default">{children}</h1>
    </ViewCenteredContainer>
  );
}
