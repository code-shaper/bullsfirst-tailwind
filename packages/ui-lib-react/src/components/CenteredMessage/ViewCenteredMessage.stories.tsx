import { Story, Meta } from '@storybook/react';
import { ViewCenteredMessage } from './ViewCenteredMessage';

export default {
  title: 'Components/ViewCenteredMessage',
  component: ViewCenteredMessage,
} as Meta;

const Template: Story = (args) => {
  return <ViewCenteredMessage>{args.children}</ViewCenteredMessage>;
};

export const ViewCenteredMessageStory = Template.bind({});
ViewCenteredMessageStory.storyName = 'ViewCenteredMessage';
ViewCenteredMessageStory.args = {
  children: 'Hello World!',
};
