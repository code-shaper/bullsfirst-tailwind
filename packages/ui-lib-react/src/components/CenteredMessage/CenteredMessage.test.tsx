import { render } from '../../test/test-utils';
import { CenteredMessage } from './CenteredMessage';

describe('<CenteredMessage />', () => {
  test('renders correctly', () => {
    const { asFragment } = render(
      <CenteredMessage>CenteredMessage</CenteredMessage>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
