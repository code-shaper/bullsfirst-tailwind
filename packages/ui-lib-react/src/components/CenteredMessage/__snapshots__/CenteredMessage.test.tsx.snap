// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CenteredMessage /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="p-6 flex-1 flex justify-center items-center text-center"
  >
    <h1
      class="text-2xl text-neutral-default"
    >
      CenteredMessage
    </h1>
  </div>
</DocumentFragment>
`;
